virttest/qemu_vm.py

```diff
@@ -146,6 +146,92 @@ def qemu_proc_term_handler(vm, monitor_exit_status, exit_status):
             error_event.error_events_bus.put(sys.exc_info())
 
 
+class _SEV(object):
+    def __init__(self, params):
+        self._secure_type = 'sev'
+        self._params = params
+        self._set_default_params()
+
+    def _set_default_params(self):
+        """
+        The following params will be set with default values:
+          vm_sev_policy: refer to libvirt, set it to 3(a sev guest)
+                         if it's not set
+        """
+        if not self._params.get('vm_sev_policy'):
+            self._params['vm_sev_policy'] = 3
+
+    def _update_sev_params(self, caps):
+        """
+        Update the following params from caps if they are not set:
+          vm_sev_cbitpos: set it from caps
+          vm_sev_reduced_phys_bits: set it from caps
+        """
+        if not self._params.get('vm_sev_cbitpos'):
+            self._params['vm_sev_cbitpos'] = caps['cbitpos']
+        if not self._params.get('vm_sev_reduced_phys_bits'):
+            self._params['vm_sev_reduced_phys_bits'] = caps['reduced-phys-bits']
+
+    @property
+    def _sectype(self):
+        return self._secure_type
+
+
+class _TDX(object):
+    def __init__(self, params):
+        self._secure_type = 'tdx'
+        self._params = params
+
+    @property
+    def _sectype(self):
+        return self._secure_type
+
+
+class VMSecDef(object):
```

Comment:
requirement(not clear) --VT translate--> Def (clear)

Reply:
private methods

Reply:
vm_params as an attribute of VM?
should params be read only?

what modified?
params -> Def, then we can update the Def, params are read only